internal/pkg/cli/svc_init.go

```diff
@@ -223,15 +226,18 @@ func (o *initSvcOpts) Execute() error {
 		}
 	}
 
-	platform, err := o.dockerEngine.RedirectPlatform(o.image)
+	detectedPlatform, targetPlatform, err := o.dockerEngine.RedirectPlatform(o.image)
 	if err != nil {
 		return fmt.Errorf("get/redirect docker engine platform: %w", err)
 	}
-	o.platform = platform
+	o.platform = targetPlatform
 	if o.platform != nil {
-		log.Warningf(`Your architecture type is currently unsupported. Setting platform %s instead.\n`, dockerengine.PlatformString(dockerengine.OSLinux, dockerengine.ArchAMD64))
-		if o.wkldType != manifest.RequestDrivenWebServiceType {
-			log.Warning("See 'platform' field in your manifest.\n")
+		// Error out early if Windows && AppRunner/RDWS.
+		if strings.Split(aws.StringValue(targetPlatform), "/")[0] == manifest.OSWindows && o.wkldType == manifest.RequestDrivenWebServiceType {
+			return errors.New("Windows is not supported for App Runner services")
+		}
+		if detectedPlatform != aws.StringValue(targetPlatform) {
+			log.Warningf("Your platform %s is currently unsupported. Setting %s instead.\nSee 'platform' field in your manifest.\n", detectedPlatform, aws.StringValue(o.platform))
```

Comment:
Why do we not return an error if the platform is not supported 🤔

Reply:
We are just informing them that while their platform isn't supported, we are still building their image but with a different platform.